src/xrpld/app/tx/detail/VaultWithdraw.cpp

```diff
@@ -288,16 +290,66 @@ VaultWithdraw::doApply()
             return ter;
     }
 
-    // Transfer assets from vault to depositor or destination account.
-    if (auto const ter = accountSend(
-            view(),
-            vaultAccount,
-            dstAcct,
-            assetsWithdrawn,
-            j_,
-            WaiveTransferFee::Yes);
-        !isTesSuccess(ter))
-        return ter;
+    if (dstAcct != account_ && !assetsWithdrawn.native())
+    {
+        bool const mptDirect = assetsWithdrawn.holds<MPTIssue>();
+        STAmount const maxSourceAmount =
+            Payment::getMaxSourceAmount(vaultAccount, assetsWithdrawn);
+        SLE::pointer sleDst = view().peek(keylet::account(dstAcct));
+        if (!sleDst)
+            return tecINTERNAL;
+
+        Payment::RipplePaymentParams paymentParams{
+            .ctx = ctx_,
+            .maxSourceAmount = maxSourceAmount,
+            .srcAccountID = vaultAccount,
+            .dstAccountID = dstAcct,
+            .sleDst = sleDst,
+            .dstAmount = assetsWithdrawn,
+            .paths = STPathSet{},
+            .deliverMin = std::nullopt,
+            .j = j_};
+
+        // If sending the assets to a different account, then this is
+        // effectively a payment. Use the Payment transaction code to call
+        // the payment engine, though only a subset of the functionality is
+        // supported in this transaction. e.g. No paths, no partial
+        // payments.
+        TER ret;
+        if (mptDirect)
+        {
+            ret = Payment::makeMPTDirectPayment(paymentParams);
+        }
+        else
+        {
+            ret = Payment::makeRipplePayment(paymentParams);
+        }
+        // Always claim a fee
+        if (!isTesSuccess(ret) && !isTecClaim(ret))
+        {
+            JLOG(j_.info())
```

Comment:
I would make it `debug()` probably; if there is indeed an easy to trigger error of this type, an `info()` level could be more easily used to flood validator logs.

Also, copy & paste error below `LoanBrokerCoverWithdraw`